crates/oxc_linter/src/config/rules.rs

```diff
@@ -121,6 +126,13 @@ impl OxlintRules {
         for rule in rules_to_replace {
             rules_for_override.replace(rule);
         }
+
+        if !rules_not_matched.is_empty() {
+            println!("The following rules do not match the currently supported rules:");
```

Comment:
Consider using `OxcDiagnostic::warn` and the reporter instead of directly printing to stdout.

Reply:
What is the reporter here? any example from code base?

Reply:
Oh ...  surfacing the warning is actually harder than I thought.